File: 2026/04-April/15-Wednesday/Solution.cs
public class Solution {
    public int ClosestTarget(string[] words, string target, int startIndex) {
        int n = words.Length;
        int ans = int.MaxValue;

        for (int i = 0; i < n; i++) {
            if (words[i] == target) {
                int d = Math.Abs(i - startIndex);
                ans = Math.Min(ans, Math.Min(d, n - d));
            }
        }

        return ans == int.MaxValue ? -1 : ans;
    }
}

File: 2026/04-April/15-Wednesday/Solution.kt
class Solution {
    fun closestTarget(words: Array<String>, target: String, startIndex: Int): Int {
        val n = words.size
        var ans = Int.MAX_VALUE
        
        for (i in 0 until n) {
            if (words[i] == target) {
                val d = Math.abs(i - startIndex)
                ans = minOf(ans, minOf(d, n - d))
            }
        }
        
        return if (ans == Int.MAX_VALUE) -1 else ans
    }
}
